src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -41,27 +43,15 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
   override def transform(dataset: Dataset[_]): DataFrame = {
     logTransform[DataFrame]({
       val invokeTextAnalyticsUdf = UDFUtils.oldUdf(invokeTextAnalytics, outputSchema)
-      val inputColNames = dataset.columns.mkString(",")
-      dataset.withColumn("Out", invokeTextAnalyticsUdf(col($(inputCol))))
-        .select(inputColNames, "Out.result.*", "Out.error.*", "Out.statistics.*", "Out.*")
-        .drop("result", "error", "statistics")
+      dataset.withColumn(getOutputCol, invokeTextAnalyticsUdf(col(getInputCol)))

```

Comment:
Use the `$(outputCol)` annotation, which is a short for `getOrDefault(param)`

Reply:
Done in new PR.